---
title: "R Notebook"
output: html_notebook
---

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*. 

Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Ctrl+Alt+I*.

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Ctrl+Shift+K* to preview the HTML file).
```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.11")
BiocManager::install("DESeq2")
BiocManager::install("apeglm")
BiocManager::install("pheatmap")
```

```{r}
library("DESeq2")

directory <- "C:/Users/Chris/Desktop/rna count"
sampleFiles <- c('trim_paired_ERR1797972_counts.txt', 'trim_paired_ERR1797973_counts.txt', 'trim_paired_ERR1797974_counts.txt', 'trim_paired_ERR1797969_counts.txt', 'trim_paired_ERR1797970_counts.txt', 'trim_paired_ERR1797971_counts.txt')
condition <- c('BH','BH','BH','Serum','Serum','Serum')

sampleTable <- data.frame(sampleName = sampleFiles,
                      fileName = sampleFiles,
                      condition = condition)

ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                   directory = directory,
                                   design= ~ condition)
ddsHTSeq
```

# Keep only rows that have at least 10 reads total
```{r}
keep <- rowSums(counts(ddsHTSeq)) >= 10
ddsHTSeq <- ddsHTSeq[keep,]
```

# Specifying the reference level
```{r}
ddsHTSeq$condition <- relevel(ddsHTSeq$condition, ref = "BH")
```

# Differential expression analysis
```{r}
dds <- DESeq(ddsHTSeq)
res <- results(dds)
res
mcols(res)$description
```
#  Log fold change shrinkage for visualization and ranking
```{r}
resultsNames(dds)
```

```{r}
resLFC <- lfcShrink(dds, coef="condition_Serum_vs_BH", type="apeglm")
resLFC
mcols(resLFC)$description
```

# Order the results table by the smallest p value
```{r}
resOrdered <- res[order(res$pvalue),]
summary(res)
sum(res$padj < 0.1, na.rm=TRUE)

write.csv(as.data.frame(resOrdered), 
          file="condition_serum_results.csv")
```

# MA Plot
```{r}
plotMA(resLFC, ylim=c(-2,2))
```

# Plot counts
```{r}
plotCounts(dds, gene=which.min(res$padj), intgroup="condition")
```

#Heatmap of the count matrix
```{r}
library("pheatmap") 

rld <- rlog(dds, blind=FALSE)

mat = assay(rld)[ head(order(res$padj),10), ]  
df = as.data.frame(colData(rld)[,c("condition")])
colnames(df) = "condition"
rownames(df) = colnames(mat)
pheatmap(mat, annotation_col=df)
```

